telemetry_test.go

```diff
@@ -14,332 +14,255 @@ import (
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-	if sc.cfg.DisableTelemetry {
-		t.Errorf("DisableTelemetry should be false. DisableTelemetry: %v", sc.cfg.DisableTelemetry)
-	}
-	if !sc.telemetry.enabled {
-		t.Errorf("telemetry should be enabled.")
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		if sct.sc.cfg.DisableTelemetry {
+			t.Errorf("DisableTelemetry should be false. DisableTelemetry: %v", sct.sc.cfg.DisableTelemetry)
+		}
+		if !sct.sc.telemetry.enabled {
+			t.Errorf("telemetry should be enabled.")
+		}
+	})
 }
 
 func funcPostTelemetryRespFail(_ context.Context, _ *snowflakeRestful, _ *url.URL, _ map[string]string, _ []byte, _ time.Duration, _ bool) (*http.Response, error) {
 	return nil, errors.New("failed to upload metrics to telemetry")
 }
 
 func TestTelemetryError(t *testing.T) {
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(context.Background(), *config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-	sr := &snowflakeRestful{
-		FuncPost:      funcPostTelemetryRespFail,
-		TokenAccessor: getSimpleTokenAccessor(),
-	}
-	st := &snowflakeTelemetry{
-		sr:        sr,
-		mutex:     &sync.Mutex{},
-		enabled:   true,
-		flushSize: defaultFlushSize,
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		sr := &snowflakeRestful{
+			FuncPost:      funcPostTelemetryRespFail,
+			TokenAccessor: getSimpleTokenAccessor(),
+		}
+		st := &snowflakeTelemetry{
+			sr:        sr,
+			mutex:     &sync.Mutex{},
+			enabled:   true,
+			flushSize: defaultFlushSize,
+		}
 
-	if err = st.addLog(&telemetryData{
-		Message: map[string]string{
-			typeKey:    "client_telemetry_type",
-			queryIDKey: "123",
-		},
-		Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
-	}); err != nil {
-		t.Fatal(err)
-	}
+		if err := st.addLog(&telemetryData{
+			Message: map[string]string{
+				typeKey:    "client_telemetry_type",
+				queryIDKey: "123",
+			},
+			Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
+		}); err != nil {
+			t.Fatal(err)
+		}
 
-	err = st.sendBatch()
-	if err == nil {
-		t.Fatal("should have failed")
-	}
+		err := st.sendBatch()
+		if err == nil {
+			t.Fatal("should have failed")
+		}
+	})
 }
 
 func TestTelemetryDisabledOnBadResponse(t *testing.T) {
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(context.Background(), *config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-	sr := &snowflakeRestful{
-		FuncPost:      postTestAppBadGatewayError,
-		TokenAccessor: getSimpleTokenAccessor(),
-	}
-	st := &snowflakeTelemetry{
-		sr:        sr,
-		mutex:     &sync.Mutex{},
-		enabled:   true,
-		flushSize: defaultFlushSize,
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		sr := &snowflakeRestful{
```

Comment:
Inline?

Reply:
Better not. It is easier to read in separation